### PR TITLE
feature: release tickets api

### PR DESCRIPTION
**Pull request by calebatwd**
A proposal pull request for adding a "releaseTickets" api to the backend service. This api is to answer the issue propose here (https://github.com/googleforgames/open-match/issues/1037) .

It essentially just issues the `DeleteTicketsFromIgnoreList` for the requested ids. This allows a director which is no longer able to handle an assignment for a match to gracefully put the tickets back into the query pool.

### CODE REVIEW

**Comment by calebatwd** [comment]
@googlebot I fixed it.

**Comment by calebatwd** [comment]
So strange. I definitely have several CLA's on file for our collaborations?

**Comment by calebatwd** [comment]
@Laremere, @yfei1. Not sure what I've done wrong but my backend_service_test failures look like I didn't set something up correctly. Will probably need some help getting it sorted :)

File: internal/app/backend/backend_service_test.go [hunk 1]
```diff
@@ -168,6 +168,132 @@ func TestDoFetchMatchesFilterChannel(t *testing.T) {
...
+			},
+			req: &pb.ReleaseTicketsRequest{
+				TicketIds: []string{"1"},
+			},
+			wantCode: codes.Unavailable,
+		},
+		{
+			description: "expect ok code when submitted list is empty",
+			preAction: func(_ context.Context, cancel context.CancelFunc, _ statestore.Service, pool *pb.Pool) {
+				cancel()
```
Comment (yfei1) [inline internal/app/backend/backend_service_test.go]: My gut tells me this test is failing because the context is cancelled before it is used ;)

Comment (calebatwd) [reply internal/app/backend/backend_service_test.go]: Ah! So silly. I'm a bit new to reading some of these lambda-setup test patterns. Thanks! I'll fix it up

**Comment by calebatwd** [comment]
I looked through some of the tutorials to try and figure out where this would make sense to add some examples, but they're all pretty optimistic. I would need to break apart the assign(matches) into something more akin to 
createGame(match)
if err -> release(match) continue
assign(match)

and maybe add some random failures to createGame?
The alternative could be to make a new sample director snippet that shows the above instead of baking it into a tutorial. Where could I put that?

**Comment by yfei1** [comment]
> I looked through some of the tutorials to try and figure out where this would make sense to add some examples, but they're all pretty optimistic. I would need to break apart the assign(matches) into something more akin to
> createGame(match)
> if err -> release(match) continue
> assign(match)
> 
> and maybe add some random failures to createGame?
> The alternative could be to make a new sample director snippet that shows the above instead of baking it into a tutorial. Where could I put that?

I think the tricky thing is the current code under the `tutorial` folder is tightly coupled with what we have in [open match tutorials](https://open-match.dev/site/docs/tutorials/matchmaker101/), so any changes to the code require a rewrite of the tutorial logics. Given that we'll be releasing 0.9 by the end of this month and we still have a reasonable amount of API changes that we wanna check into the release, I propose adding a `How to handle errors` or `FAQ` tab under the current tutorials. WDYT?

**Comment by calebatwd** [comment]
> > I looked through some of the tutorials to try and figure out where this would make sense to add some examples, but they're all pretty optimistic. I would need to break apart the assign(matches) into something more akin to
> > createGame(match)
> > if err -> release(match) continue
> > assign(match)
> > and maybe add some random failures to createGame?
> > The alternative could be to make a new sample director snippet that shows the above instead of baking it into a tutorial. Where could I put that?
> 
> I think the tricky thing is the current code under the `tutorial` folder is tightly coupled with what we have in [open match tutorials](https://open-match.dev/site/docs/tutorials/matchmaker101/), so any changes to the code require a rewrite of the tutorial logics. Given that we'll be releasing 0.9 by the end of this month and we still have a reasonable amount of API changes that we wanna check into the release, I propose adding a `How to handle errors` or `FAQ` tab under the current tutorials. WDYT?

I like it. What's the process here. Do I need more tests or do I need to complete documentation and FAQs for the new api? Or do we checkin and do subsequent pr with the description? It's going to be a week before I can get to doing any tutorial work for this api.

File: api/backend.proto [hunk 2]
```diff
@@ -110,4 +118,14 @@ service Backend {
       body: "*"
     };
   }
+
+  // ReleaseTickets removes the submitted tickets from the list that prevents tickets 
+  // that are awaiting assignment from appearing in MMF queries, effectively putting them back into
+  // the matchmaking pool
```
Comment (Laremere) [inline api/backend.proto]: Please add the following warning:
```
BETA FEATURE WARNING:  This call and the associated Request and Response messages are
not finalized and still subject to possible change or removal.
```

**Pull request merged**